documentation/guide-configuration.asciidoc

```diff
@@ -156,47 +156,189 @@ We recommend the following base layout for the hierarchical business configurati
+Let's create service bean class to retrieve the values from `encrypted.properties` file.
+The decrypted value can be retrieved using the `@Value` annotation or the getProperty() method of `Environment` class. See below example:
+
+[source,java]
+----
+@Service
+public class PropertyMapperJasyptStarter {
+
+    @Value("${encrypted.property}")
+    private String property;
+
+    public String getProperty() {
+        return property;
+    }
+
+    public String getPasswordUsingEnvironment(Environment environment) {
+        return environment.getProperty("encrypted.property");
+    }
+}
+----
+
+Using the above service class and setting the secret key which we used for encryption, we can easily retrieve the decrypted password and use in our application.
+
+==== How to use Jasypt-SpringBoot
+For projects not using `@SpringBootApplication` or `@EnableAutoConfiguration`, we can use the jasypt-spring-boot dependency directly
+
+[source,xml]
+----
+<dependency>
+    <groupId>com.github.ulisesbocchio</groupId>
+    <artifactId>jasypt-spring-boot</artifactId>
+    <version>2.0.0</version>
+</dependency>
+----
+
+Please check for latest version link:https://mvnrepository.com/artifact/com.github.ulisesbocchio/jasypt-spring-boot[here] .
+
+let's encrypt the text `Password@2` with secret key `password` and add it to the `encrypted.properties`.
+
+[source,properties]
+----
+encrypted.property=ENC(dQWokHUXXFe+OqXRZYWu22BpXoRZ0Drt)
+----
+
+New configuration class is required for `Jasypt-springboot` dependency and we need to add the annotation `@EncryptablePropertySource` at the class level.
+
+Below is an example,
+
+[source,java]
+----
+@Configuration
+@EncryptablePropertySource("encrypted.properties")
+public class AppJasyptConfig {
+}
+----
+
+And a service class to map the property in encrypted.properties.
+[source,java]
+----
+@Service
+public class PropertyMapperJasypt {
+
+    @Value("${encrypted.property}")
+    private String property;
+
+    public String getProperty() {
+        return property;
+    }
+}
+----
+Using the above service class and setting the secret key which we used for encryption, we can easily retrieve the encrypted.property.
+
+==== How to use custom Jasypt Encryptor
+
+The encryptors mentioned above has their own configuration. We can also define our own Jasypt encryptor and use it on our application.
+
+See below example,
+
+[source,java]
+----
+@Bean(name = "encryptorBean")
+public StringEncryptor stringEncryptor() {
+    PooledPBEStringEncryptor encryptor = new PooledPBEStringEncryptor();
+    SimpleStringPBEConfig config = new SimpleStringPBEConfig();
+    config.setPassword("password");
+    config.setAlgorithm("PBEWithMD5AndDES");
+    config.setKeyObtentionIterations("1000");
+    config.setPoolSize("1");
+    config.setProviderName("SunJCE");
+    config.setSaltGeneratorClassName("org.jasypt.salt.RandomSaltGenerator");
+    config.setStringOutputType("base64");
+    encryptor.setConfig(config);
+    return encryptor;
+}
```

Comment:
So are these the parameters that make the encrpytion secure according to state of the art?
Did you align this with @maybeec or where did you get this code from?

Reply:
No i have not discussed regarding this with @maybeec but this is an example i learned from a tutorial.

Reply:
This config allows you to adapt the algorithm.
Anyhow the algorithm used here is the standard one which is quite insecure. Anyhow, at the time I tried making it more secure I was not able to get any other algorithm to work for any reason. Possibly this has been fixed meanwhile!?